### PR TITLE
Fix back links on case attachment pages

### PR DESCRIPTION
**Pull request by slorek**
## Description
This is just a crude sticky plaster solution to fix the back links on the add/edit/remove attachment pages for a case, so that users don't see error messages. It should also resolve some exceptions we're seeing in Sentry due to the links routing to the wrong controller (which is part of the creation flow).

The changed page templates are shared for investigations/businesses/products, but there are big differences in how these three view template groups and routes work.

Businesses/products do not implement proper routes i.e. /products/:id/attachments, instead they implement a Javascript tab only, so the back links go to the overview page instead. It would be possible to add a further sticky plaster for investigations to go to `/attachments`, but I think if we want to go further we should just fix the crazy routing design.

The design really is crazy: the model association name for attachments is `documents`, but the model name for a document is resolved to `attachment` by the polymorphic route helpers. The route name is `documents`, so it is not resolved correctly, meaning it has to be overridden using [custom helpers](https://github.com/UKGovernmentBEIS/beis-opss-psd/blob/master/psd-web/app/helpers/url_helper.rb).

Except the route name for *cases* is `attachments` on the `index` action only; member actions are called documents.  But the index action name on products/businesses is `documents`.

All of this means we have some crude URL helpers to maintain, which I've modified here.

There's plenty of other WTF in the attachments routing and behaviour, so to improve further we should consider rewriting this. The scope of this change is just to get the links working for users.

## Checklist:

- [x] Automated checks are passing locally.
- [ ] CHANGELOG updated if change is worth telling users about.
### General testing
- [ ] Test without javascript
- [ ] Test on small screen
### Accessibility testing
- [ ] Works keyboard only
- [ ] Tested with one screen reader
- [ ] Zoom page to 400% - content still visible
- [ ] Disable css - does content make sense and appear in a logical order?
- [ ] Passes automated checker (automated in build or manual)

### CODE REVIEW

File: psd-web/spec/features/add_attachment_spec.rb [hunk 1]
```diff
@@ -69,11 +69,13 @@
   def expect_to_be_on_add_attachment_page
     expect(page).to have_current_path("/cases/#{investigation.pretty_id}/documents/new/upload")
     expect(page).to have_selector("h1", text: "Add attachment")
+    expect(page).to have_link("Back", href: "/cases/#{investigation.pretty_id}")
```
Comment (StupidCodeFactory) [inline psd-web/spec/features/add_attachment_spec.rb]: Didn't notice this before, but I'd prefer to use the routes helper. `investigation_path(investigation)` and so on.

**Comment by nasirkhanbeis** [comment]
@slorek can't test back link on edit/remove attachment pages as I see processing image label but not the actual image
image

**Comment by edwardhorsford** [comment]
I created my own project and uploaded a file. When viewing the attachments tab, the edit / view / remove links are not present - I don't know if this is related to the image being stuck in processing or not.

Screenshot 2020-01-02 at 12 28 33

**Comment by slorek** [comment]
Missing edit/remove links on review app were resolved by https://github.com/UKGovernmentBEIS/beis-opss-psd/pull/125

**Pull request merged**